x-pack/plugins/ml/common/util/errors.ts

```diff
@@ -17,3 +18,35 @@ export function getErrorMessage(error: any) {
 
   return JSON.stringify(error);
 }
+
+export const extractErrorMessage = (
```

Comment:
this function contains plenty of `if` statements, worth to covert with unit tests and provide a better typing

Reply:
I'll do this an a follow up PR